src/Hypergrid.js

```diff
@@ -1257,7 +1258,7 @@ var Hypergrid = Base.extend('Hypergrid', {
 
                 // target is to right of scrollable columns; positive delta scrolls right
                 // Note: The +1 forces right-most column to scroll left (just in case it was only partially in view)
-                (delta = c - dw.origin.x + 1) > 0
```

Comment:
Done! :p
Pull the last changes (2 commit I think...) and test it again please.

I've updated the scrollToMakeVisible() code so that, only when it needs to scroll right, it calculates each columns left and right positions, until reaching the target column's right position. Knowing this, we then calculate the available viewport dimensions (without any fixed columns and headers width) and check which is the minimal index that we can scroll into, that will still show the target column fully inside the viewport. 
I've done some tests with multiple column's configurations (e.g. widths, fixed position, without numbers column) and looks good.

Hope it works!
Thanks!